pizza_cutter/des_pizza_cutter/_slice_data.py

```diff
@@ -0,0 +1,181 @@
+    msk = (bm_rot & bad_flags) != 0
+    if np.any(msk):
+        bmask[msk] |= bm_rot[msk]
+
+
+def _interp_image(*, image, good_msk, bad_msk, yx):
+    img_interp = CloughTocher2DInterpolator(
+        yx[good_msk, :],
+        image[good_msk],
+        fill_value=0.0)
+    interp_image = image.copy()
+    interp_image[bad_msk] = img_interp(yx[bad_msk, :])
+    return interp_image
+
+
+def _draw_noise_image(*, weight, rng):
+    return rng.normal(size=weight.shape) * np.sqrt(1.0/weight)
+
+
+def interpolate_image_and_noise(*, image, weight, bmask, bad_flags, rng):
+    """Interpolate an image using the
+    `scipy.interpolate.CloughTocher2DInterpolator`. An interpolated noise
+    field is returned as well.
+
+    Parameters
+    ----------
+    image : array-like
+        The image to interpolate.
+    weight : array-like
+        The weight map of the image to interpolate.
+    bmask : array-like
+        The bit mask for the slice.
+    bad_flags : int
+        Pixels with in the bit mask using
+        `(bmask & bad_flags) != 0`.
+    rng : `numpy.random.RandomState`
+        An RNG instance to use.
+
+    Returns
+    -------
+    interp_image : array-like
+        The interpolated image.
+    interp_weight : array-like
+        The interpolated weight map.
+    """
+    bad_msk = (weight <= 0) | ((bmask & bad_flags) != 0)
+
+    if np.any(bad_msk):
+        good_msk = ~bad_msk
+
+        # ravel it all here
+        # eveything past here is done on the raveled image
+        good_msk = good_msk.ravel()
+        bad_msk = bad_msk.ravel()
+
+        # we need the coordinates for the image interpolator
+        y, x = np.mgrid[0:image.shape[0], 0:image.shape[1]]
+        x = x.ravel()
+        y = y.ravel()
+        yx = np.zeros((x.size, 2))
+        yx[:, 0] = y
+        yx[:, 1] = x
+
+        # now the real work begins
+        interp_image = _interp_image(
+            image=image.ravel(),
+            good_msk=good_msk,
+            bad_msk=bad_msk,
+            yx=yx)
+
+        # fill the weight map with the median so we can draw a noise map
+        # we could apply the interpolator too?
+        interp_weight = weight.copy().ravel()
+        interp_weight[bad_msk] = np.median(interp_weight[good_msk])
+
+        # now draw a noise map and apply an interp to it
+        # this is to propagate how the interpolation correlates pixel noise
+        # so it has to be done to the noise map
+        noise = _draw_noise_image(weight=interp_weight, rng=rng)
+        interp_noise = _interp_image(
+            image=noise,
+            good_msk=good_msk,
+            bad_msk=bad_msk,
+            yx=yx)
+
+        # now unravel
+        interp_image = interp_image.reshape(image.shape)
+        interp_noise = interp_noise.reshape(image.shape)
+
+        return interp_image, interp_noise
+    else:
+        # return a copy here since the caller expects new images
+        return image.copy(), _draw_noise_image(weight=weight, rng=rng)
```

Comment:
should we still use median of weight map in this case?

Reply:
yes good catch!

Reply:
Hrmmmmmm. Actually I don't think so. I think we only fill with the median when we have a bad pixel. Above in the other branch we use the reported value for any good pixels. So I think we don't need to.

Reply:
these areas will be treated differently if the weight map is not constant

Reply:
Ahhhh wait. I think I am not understanding your comment. 

For a masked pixel, do we need to muck with the weight map at all? I think we can just use it as is right? 

Right now for any pixel that needs interpolation in the image, I am replacing the weight map for that pixel by the median. I don't think we need to do that step?

Reply:
Note I am only replacing the weight map by its median in the masked pixels, not the full thing. So any pixels that are not masked have the same weight in either branch of the if statement.

Reply:
You should make sure those pixels don't have some weight map value set for them.  I don't remember if they do at times.